src/core/components/progress/progress.tsx

```diff
@@ -23,27 +23,42 @@ const responsiveSizeClassNames: ResponsiveAttributeClassMap<ProgressSize> = {
     },
 };
 
+const variantToClassNames: Record<ProgressVariant, string> = {
+    primary: 'bg-primary-400',
+    neutral: 'bg-neutral-400',
+    success: 'bg-success-400',
+    critical: 'bg-critical-500',
+};
+
 export const Progress: React.FC<IProgressProps> = (props) => {
-    const { value, size = 'md', responsiveSize, className, ...otherProps } = props;
+    const { value, size = 'md', responsiveSize, className, variant = 'primary', indicator, ...otherProps } = props;
 
     const processedValue = Math.min(Math.max(1, value), 100);
-
+    const processedIndicator = indicator && Math.min(Math.max(1, indicator), 100);
+    const indicatorHeight = size === 'md' ? 'h-4' : 'h-2';
```

Comment:
Good catch, I didn't think about the different sizes. But we need to make this style responsive as we're already doing for the container height